test/shadowing.jl

```diff
@@ -24,17 +24,89 @@ using Test
     sol_attractor = solve(prob_attractor,Vern9(),abstol=1e-14,reltol=1e-14)
 
     g(u,p,t) = u[end]
+    function dg(out,u,p,t,i)
+      fill!(out, zero(eltype(u)))
+      out[end] = -one(eltype(u))
+    end
     lss_problem1 = ForwardLSSProblem(sol_attractor, ForwardLSS(), g)
+    lss_problem1a = ForwardLSSProblem(sol_attractor, ForwardLSS(), nothing, dg)
     lss_problem2 = ForwardLSSProblem(sol_attractor, ForwardLSS(alpha=DiffEqSensitivity.Cos2Windowing()), g)
+    lss_problem2a = ForwardLSSProblem(sol_attractor, ForwardLSS(alpha=DiffEqSensitivity.Cos2Windowing()), nothing, dg)
     lss_problem3 = ForwardLSSProblem(sol_attractor, ForwardLSS(alpha=10), g)
+    lss_problem3a = ForwardLSSProblem(sol_attractor, ForwardLSS(alpha=10), g, dg) #ForwardLSS with time dilation requires knowledge of g
+
+    adjointlss_problem = AdjointLSSProblem(sol_attractor, AdjointLSS(alpha=10.0), g)
+    adjointlss_problem_a = AdjointLSSProblem(sol_attractor, AdjointLSS(alpha=10.0), g, dg)
+
+    res1 = DiffEqSensitivity.__solve(lss_problem1)
```

Comment:
why testing on the lower level?

Reply:
yeah, probably  `__solve` should be renamed https://github.com/SciML/DiffEqSensitivity.jl/blob/79a727b409799025f0c05c20be75f228f6d2505f/src/lss.jl#L296. There is no higher level `solve(LSSProblem)` at the moment. `__solve` is the equivalent to `adjoint_sensitivities()` for an `AdjointODEProblem`.  I had it as `solve()` before and changed it in https://github.com/SciML/DiffEqSensitivity.jl/pull/420/commits/85539e8934bc1c8d02239fdd24d8f03b4783e791 [associated comment: https://github.com/SciML/DiffEqSensitivity.jl/pull/420#discussion_r649537782] because of the error I got.